### PR TITLE
Don’t activate a transport that isn’t connected

### PR DESCRIPTION
**Pull request by SimonWoolf**
Fixes https://github.com/ably/ably-js/issues/247

### CODE REVIEW

File: common/lib/transport/connectionmanager.js [hunk 1]
```diff
@@ -422,6 +422,14 @@ var ConnectionManager = (function() {
 		if(existingState == this.states.closing || existingState == this.states.closed)
 			return false;
 
+		/* if the transport is not connected (eg because it failed during a
+		 * scheduleTransportActivation#onceNoPending wait) then don't activate it */
+		if(!transport.isConnected) {
+			Logger.logAction(Logger.LOG_MINOR, 'ConnectionManager.activateTransport()', 'Declining to activate transport ' + transport + ' since it appears to no longer be connected');
+			Utils.arrDeleteValue(this.pendingTransports, transport);
```
Comment (paddybyers) [inline common/lib/transport/connectionmanager.js]: I think I would have reversed the order of these things; ie unconditionally remove from the `pendingTransports` list, then  decline to activate + return if not connected.

**Comment by paddybyers** [comment]
LGTM but see comment above.

File: common/lib/transport/connectionmanager.js (hunk 1 above)
Comment (SimonWoolf) [inline common/lib/transport/connectionmanager.js]: good point

**Comment by SimonWoolf** [comment]
Merged as c2a4912

**Pull request closed without merging**